hack/run_in_cloud.sh

```diff
@@ -0,0 +1,75 @@
+#!/bin/bash
+# Note: Tested only with buildah / podman
+
+set -eux
+
+DIR=$( cd -- "$( dirname -- "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )
+
+if [ -z "${CNCC_REPOSITORY}" ]; then
+	echo "Please set environment variable CNCC_REPOSITORY and point it to your container repository"
+	exit 1
+fi
+
+CONTAINER_ENGINE="${CONTAINER_ENGINE:-podman}"
+BUILDAH_BIN="buildah"
+BUILDAH_BUILD_CMD="buildah bud"
+PODMAN_BIN="podman"
+PODMAN_CMD="podman"
+if [ "$CONTAINER_ENGINE" == "docker" ]; then
+	BUILDAH_BIN="docker"
+	PODMAN_BIN="docker"
+	BUILDAH_BUILD_CMD="docker build"
+	PODMAN_CMD="docker"
+fi
+
+for command in mktemp awk oc uuidgen $BUILDAH_BIN $PODMAN_BIN; do
+	if ! command -v $command &> /dev/null
+	then
+		echo "$command not found"
+		exit 1
+	fi
+done
+
+UUID=$(uuidgen)
+IMAGE="${CNCC_REPOSITORY}:${UUID}"
+
+# retry logic for "Bad Gateway message" encountered a lot during testing
+push_container() {
+	f=$(mktemp)
+	for i in 1 2 3; do
+		$PODMAN_CMD push ${IMAGE} 2>&1 | tee $f
+		if [ ${PIPESTATUS[0]} -ne 0 ]; then
+			if grep -q "502 Bad Gateway" $f; then
+				echo "Got 502 Bad Gateway error, this was try: $i/3"
```

Comment:
you should not get these unless there is some sort of infrastructural error, in which case there is not reason to think that trying 3 times should help. Just push it, and if it fails, then bail out. The user can try again if they think there is reason to try again.

Reply:
done